daphne/src/lib.rs

```diff
@@ -236,6 +242,35 @@ pub struct ProblemDetails {
     pub detail: Option<String>,
 }
 
+/// DAP version used for a task.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub enum DapVersion {
+    #[serde(rename = "v01")]
+    Draft01,
+
+    #[serde(other)]
+    Unknown,
+}
+
+impl DapVersion {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            DapVersion::Draft01 => "v01",
+            DapVersion::Unknown => "unknown",
+        }
+    }
+}
+
+impl From<&str> for DapVersion {
+    fn from(version: &str) -> Self {
+        if version == DapVersion::Draft01.as_str() {
```

Comment:
A `match` statement would be cleaner here.
```rust
match version {
   "v01" => DapVersion::Draft01,
    _ => DapVersion::Unknown,
}
```
That way you can do away with `as_str()` above altogether.

Reply:
Agreed, this is cleaner.